File: ch.jnetwork.sac-backend/Api/Model/Browser/BrowserNode.cs
namespace SimpleAngularControls.Api.Model.Browser
{
    public class BrowserNode : IBrowserNode
    {
        public string Name { get; set; } = string.Empty;

        public List<IBrowserNode> ChildNodes { get; set; } = new List<IBrowserNode>();

        public List<IBrowserFile>? Files { get; set; } = null;

        IEnumerable<IBrowserNode> IBrowserNode.ChildNodes => this.ChildNodes;

        IEnumerable<IBrowserFile>? IBrowserNode.Files => this.Files;
    }
}

File: ch.jnetwork.sac-controls/projects/jngcontrols-common/src/components/browser/models/browsernode.ts
import { IBrowserFile } from './browserfile';

/**
 * Interface für Node in File Browser Component
 */
export interface IBrowserNode {
  /**
   * Name des Nodes
   */
  Name: string;
  /**
   * Child Nodes
   */
  ChildNodes: IBrowserNode[];
  /**
   * File Items in diesem Node
   */
  Files: IBrowserFile[];

  /**
   * Pfad für diesen Node
   */
  Path: string;
  /**
   * Node ist ausgeklappt
   */
  IsExpanded: boolean;
  /**
   * Node wird bearbeitet
   */
  IsEditMode: boolean;
  /**
   * Node wurde erzeugt aber noch nicht gespeichert
   */
  IsNewNode: boolean;
}
